cmd/xconn/main.go

```diff
@@ -1,24 +1,125 @@
 package main
 
 import (
+	"bytes"
+	_ "embed" // nolint:gci
+	"fmt"
 	"log"
+	"os"
+
+	"github.com/alecthomas/kingpin/v2"
+	"golang.org/x/exp/slices"
+	"gopkg.in/yaml.v3"
 
 	"github.com/xconnio/wampproto-protobuf/go"
 	"github.com/xconnio/xconn-go"
 )
 
-func main() {
-	router := xconn.NewRouter()
-	router.AddRealm("realm1")
+var (
+	//go:embed config.yaml.in
+	sampleConfig []byte
+)
+
+const (
+	versionString = "0.1.0"
+
+	ConfigDir  = ".xconn"
+	ConfigFile = ConfigDir + "/config.yaml"
+
+	ProtobufSubProtocol = "wamp.2.protobuf"
+)
+
+type cmd struct {
+	parsedCommand string
+
+	init *kingpin.CmdClause
 
-	server := xconn.NewServer(router, nil)
+	start *kingpin.CmdClause
+}
+
+func parseCommand(args []string) (*cmd, error) {
+	app := kingpin.New(args[0], "XConn")
+	app.Version(versionString).VersionFlag.Short('v')
 
-	serializer := &wampprotobuf.ProtobufSerializer{}
-	protobufSpec := xconn.NewWSSerializerSpec("wamp.2.protobuf", serializer)
+	c := &cmd{
+		init:  app.Command("init", "Initialize sample router config."),
+		start: app.Command("start", "Start the router."),
+	}
+
+	parsedCommand, err := app.Parse(args[1:])
+	if err != nil {
+		return nil, err
+	}
+	c.parsedCommand = parsedCommand
+
+	return c, nil
+}
+
+func Run(args []string) (string, error) {
+	c, err := parseCommand(args)
+	if err != nil {
+		return "", err
+	}
 
-	if err := server.RegisterSpec(protobufSpec); err != nil {
-		log.Fatal(err)
+	switch c.parsedCommand {
+	case c.init.FullCommand():
+		if err := os.MkdirAll(ConfigDir, os.ModePerm); err != nil {
+			return "", err
+		}
+
+		if err = os.WriteFile(ConfigFile, sampleConfig, 0600); err != nil {
+			return "", fmt.Errorf("unable to write config: %w", err)
+		}
+
+		return fmt.Sprintf("Sample router config initialized at %s.", ConfigFile), nil
```

Comment:
```suggestion
		return fmt.Sprintf("Sample router config initialized at %s.", ConfigFile), nil
```
i think this one should not return a string, it should only return error. This is different because we will be running the router as well, in that case returning output doesnt make sense